src/lib/logstore/log_store_service.cpp

```diff
@@ -106,7 +106,7 @@ void LogStoreService::start(bool format) {
 }
 
 void LogStoreService::stop() {
-    device_truncate(nullptr, true, false);
+    // device_truncate(nullptr, true, false);
```

Comment:
this device_truncate is calling into log_dev to do actual truncate, we should not do that without setting the compact_lsn in logstore (through nuraft::compact) because only nuraft knows which should be the correct compact_lsn. We already persisted the compact_lsn in superblock, in reboot, it can catch up in next timer schedule or next critical alert is hit. I don't quite get what is the 'crunch issue' here in your comment.